amethyst_network/src/network_socket.rs

```diff
@@ -77,41 +80,49 @@ where
             //rx1,tx2
             let send_queue = rx1;
             let receive_queue = tx2;
-            let socket = socket;
+            let mut socket = socket;
 
             'outer: loop {
                 // send
                 for control_event in send_queue.try_iter() {
                     match control_event {
                         InternalSocketEvent::SendEvents { target, events } => {
                             for ev in events {
-                                send_event(&ev, &target, &socket);
+                                send_event(&ev, &target, &mut socket);
                             }
                         }
                         InternalSocketEvent::Stop => break 'outer,
                     }
                 }
 
                 // receive
-                let mut buf = [0 as u8; 2048];
                 loop {
-                    match socket.recv_from(&mut buf) {
+                    match socket.recv() {
                         // Data received
-                        Ok((amt, src)) => {
-                            if let Err(_) = receive_queue.send(RawEvent {
-                                byte_count: amt,
-                                data: buf[..amt].to_vec(),
-                                source: src,
-                            }) {
+                        Ok(Some(packet)) => {
+                            let payload = packet.payload();
+                            let raw_event = RawEvent {
+                                byte_count: payload.len(),
+                                data: payload.to_owned(),
+                                source: packet.addr(),
+                            };
+
+                            if let Err(_) = receive_queue.send(raw_event) {
                                 error!("`NetworkSocketSystem` was dropped");
                                 break 'outer;
                             }
-                        }
+                        },
+                        Ok(None) => { info!("Received data, but awaiting still other data.") }
```

Comment:
Do we really need to log this?

Reply:
It's up to you, we could just ignore it.

Reply:
I'll remove it.